sample/buildSrc/build.gradle

```diff
@@ -0,0 +1,21 @@
+repositories {
+    jcenter()
+}
+
+apply plugin: 'groovy'
+
+dependencies {
+    compile gradleApi()
+}
+
+sourceSets {
```

Comment:
The old approach was doing the same, linking via buildSrc, imho.

The problem with IntelliJ/AS is with the different Gradle versions between the sample and the plugin. I think it's important not to lock the Gradle version of the plugin with the one from the sample, at least as long as we do not have a proper testing set-up to ensure compatibility.

Opening the sample as a separate project might be a little inconvenient, but it's not that big of a deal, imho.